src/agenda/index.js

```diff
@@ -216,13 +216,21 @@ export default class AgendaView extends Component {
     this._isMounted = false;
   }
 
-  componentWillReceiveProps(props) {
-    if (props.items) {
-      this.setState({
-        firstResevationLoad: false
-      });
+  componentWillReceiveProps(nextProps) {
+    const nextState = {};
+    if (nextProps.items) {
+      nextState.firstResevationLoad = false;
+    }
+    if (
+      nextProps.selected &&
```

Comment:
What if it changed from being selected to not selected? No update?

Reply:
It's been a while since I opened this PR so I'm not really in context. Unless I'm mistaken:

Changes in the optional ``selected`` trigger a change on the internal ``selectedDay`` (well, they actually don't and that's what the PR is trying to fix). This internal date is not optional so even if the prop changes to undefined we can't set the internal date to undefined (hence the check).

Reply:
What I mean is that you pass the condition only if `selected === true`. But what if it was true but now it's false?

Reply:
``selected``is **not** a boolean (I know it sounds like it is, I certainly wouldn't had named it like that). ``selected``is a date. What I am checking is that it has a value, as we cannot assign undefined or null to ``selectedDay``; we cannot unselect the date on the component (we get an exception otherwise).